source/particle/property/interface.cc

```diff
@@ -824,10 +823,8 @@ namespace aspect
                        "all") != this->plugin_names.end())
           {
             this->plugin_names.clear();
-            for (typename std::list<typename aspect::internal::Plugins::PluginList<aspect::Particle::Property::Interface<dim>>::PluginInfo>::const_iterator
-                 p = std::get<dim>(registered_plugins).plugins->begin();
-                 p != std::get<dim>(registered_plugins).plugins->end(); ++p)
-              this->plugin_names.push_back (std::get<0>(*p));
+            for (const auto &p : std::get<dim>(this->registered_plugins))
```

Comment:
This should be:
```suggestion
            for (const auto &p : std::get<dim>(registered_plugins).plugins)
```

You stumbled into maybe the most complicated data container in ASPECT here ;-)